### PR TITLE
Add BSCodec implementation and recipe

### PR DESCRIPTION
**Pull request by whr-a**
## What did you change?

* Added the core model implementation for BSCodec under `espnet2/gan_codec/bscodec`.
* Added the corresponding recipe (including training, inference, and evaluation) in `egs2/bscodec/codec1`.
* Included a README with model results, guidelines for usage, and the paper citation.

## Why did you make this change?

This PR introduces **BSCodec**, a new band-split neural codec. As shown in the paper and results, this model performs well across multiple domains (speech, sound, music) and achieves better reconstruction performance at lower bitrates compared to previous general-purpose codecs.

## Additional Context

* **Paper:** [`arXiv preprint`](https://arxiv.org/abs/2511.06150) (The citation is in the README)
* **Pretrained Models:** Available at [Hugging Face](https://huggingface.co/anonymous-release/BSCodec/tree/main)

### CODE REVIEW

File: egs2/bscodec/codec1/run.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+fi
+
+
+train_set=train_all
+valid_set=dev_sub
+test_sets="test_all"
+
+model=BSCodec_band_vq_5band
+
+train_config=conf/tuning/pretrain_gan/${model}.yaml
```
[the remark this answers is not in the document]
Comment (whr-a) [reply egs2/bscodec/codec1/run.sh]: Fixed.

**Comment by sw005320** [comment]
There are many CI issues.
https://github.com/espnet/espnet/actions/runs/19319439898/job/55257666585?pr=6297
Please fix them.

**Comment by sw005320** [comment]
@ftshijt, can you review this PR?

**Comment by sw005320** [comment]
@whr-a, it still has a lot of CI errors

**Comment by whr-a** [comment]
OK, I'll check them.

File: egs2/bscodec/codec1/conf/scoring/score.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# codec example yaml config
```
Comment (ftshijt) [inline egs2/bscodec/codec1/conf/scoring/score.yaml:1]: It could be better to clean the config setup a bit

File: espnet2/gan_codec/shared/quantizer/modules/core_vq.py [hunk 3]
```diff
@@ -593,3 +593,58 @@ def decode(self, q_indices: torch.Tensor) -> torch.Tensor:
             quantized = layer.decode(indices)
             quantized_out = quantized_out + quantized
         return quantized_out
+
+
+class BandVectorQuantization(nn.Module):
```
Comment (ftshijt) [inline espnet2/gan_codec/shared/quantizer/modules/core_vq.py]: Maybe inheritance from the original class would be easier, considering many shared components and almost same behavior except for the band-related loss

File: egs2/bscodec/codec1/run.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+inference_config=conf/decode.yaml
+score_config=conf/score.yaml
+
+./codec.sh \
+    --local_data_opts "--trim_all_silence false" \
+    --fs ${fs} \
+    --ngpu 1 \
+    --nj 64\
+    --stage 6\
+    --stop_stage 6\
```
Comment (ftshijt) [inline egs2/bscodec/codec1/run.sh]: Please clean the setup at by here

**Review by ftshijt — COMMENTED** [review]
Please also register the one by egs2/README.md;

File: egs2/bscodec/codec1/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# BSCodec
+BSCodec is a codec that performs well across multiple domains, offering relatively lower bit rates and better reconstruction performance compared to previous general-purpose codecs.
+
+It is based on DAC structure and band-split strategy, and was trained using the ESPnet codec training pipeline.
+
+The model checkpoint is available at https://huggingface.co/anonymous-release/BSCodec/tree/main
```
Comment (ftshijt) [inline egs2/bscodec/codec1/README.md]: We can switch to public ones and also add the citation to the paper

**Comment by sw005320** [comment]
@whr-a, please reflect the review comments

**Comment by whr-a** [comment]
Ok, I'll fix them soon.
